clang/lib/Format/TokenAnnotator.cpp

```diff
@@ -148,6 +160,24 @@ class AnnotatingParser {
     }
   }
 
+  const FormatStyle::FunctionDeclarationWithKeywords *
+  isInsideFunctionWithKeywordedParameters(const FormatToken &Token) const {
+    const FormatToken *Previous = &Token;
+    while (auto Prev = Previous->getPreviousNonComment())
+      Previous = Prev;
```

Comment:
`Line.First`?

Reply:
I hardly understand what am I doing here, sorry. In particular, I don't understand which pass (is that settings-dependent?) splits consecutive macros in the same line into multiple lines? May I rely on that behaviour?

Reply:
Ah, so `AnnotatedLine` contains a single statement only?

Reply:
Something like that.

Reply:
Thanks! Then can I look for the first `tok::l_paren` in the `Line` and take previous non-comment from that?